lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/gson/AbstractTypeTypeAdapter.java

```diff
@@ -30,18 +30,35 @@
 import java.io.IOException;
 import java.util.Map;
 
-public abstract class AbstractTypeAdapterFactory<T> extends TypeAdapter<T> {
+/**
+ * A type adapter which creates an object of a specific type based on a hidden "type" field.
+ * An implementation of this abstract class must specify which type name translates to which target class.
+ *
+ * @param <T> the base type of which objects are created during deserialization
+ */
+public abstract class AbstractTypeTypeAdapter<T> extends TypeAdapter<T> {
```

Comment:
This name seems a little bit funny to me maybe we can find a better name.

Reply:
The old name was definitely wrong, as it was no Factory. I want to end the class name with `TypeAdapter`, as it implements that interface. We could name it `TypeFieldTypeAdapter`, which is not much better but at least it looks intentional.